ci/jenkins_status.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+
+"""Parse Jenkins build test results"""
+
+import argparse
+import json
+import sys
+import urllib
+from urllib.request import urlopen
+
+JENKINS_HOME = "https://build.hpdd.intel.com/job/daos-stack"
+
+
+class TestResult:
+    """Represents a single Jenkins test result"""
+
+    def __init__(self, data, blocks):
+        name = data["name"]
+        if "-" in name:
+            try:
+                (num, full) = name.split("-", 1)
+                test_num = int(num)
+                if full[-5] == "-":
+                    full = full[:-5]
+                name = f"{full} ({test_num})"
+            except ValueError:
+                pass
+        self.name = name
+        self.cname = data["className"]
+        self.skipped = False
+        self.passed = False
+        self.failed = False
+        assert data["status"] in ("PASSED", "FIXED", "SKIPPED", "FAILED", "REGRESSION")
```

Comment:
Maybe better to gracefully handle?

Reply:
For an internal script I'm OK with this, if there is another status then we want to know about it so we handle it properly.  As we're scraping Jenkins then we can stress-test this code against previous builds.

Reply:
The problem with `assert` is that the entire script will fail without much debug info.